### PR TITLE
IN-30 feat: add color palette

### PR DESCRIPTION
**Pull request by vitaliiRss**

### CODE REVIEW

File: src/components/colorPalette/colorPalette.module.scss [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+.color {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+
+  width: calc((100% / 5) - 16px);
+
+  // min-width: 200px;
```
Comment (SergFediaev) [inline src/components/colorPalette/colorPalette.module.scss:38]: Commented style.

File: src/components/colorPalette/colorPalette.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import type { Meta, StoryObj } from "@storybook/react"
+
+import { type ColorGroup, ColorPalette } from "./colorPalette"
+
+const meta: Meta<typeof ColorPalette> = {
```
Comment (SergFediaev) [inline src/components/colorPalette/colorPalette.stories.tsx:5]: I think one type is enough here:

`
const meta = {
...
} satisfies Meta<typeof ColorPalette>
`

File: src/components/colorPalette/colorPalette.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  component: ColorPalette,
+  title: "01. Design System/Colors"
+} satisfies Meta<typeof ColorPalette>
+
+export default meta
+
+type Story = StoryObj<typeof ColorPalette>
```
Comment (SergFediaev) [inline src/components/colorPalette/colorPalette.stories.tsx:12]: Here you can use a meta as a type:

`
type Story = StoryObj<typeof meta>
`

**Review by SergFediaev — CHANGES_REQUESTED** [review]

File: src/components/colorPalette/colorPalette.stories.tsx (hunk 2 above)
Comment (vitaliiRss) [reply src/components/colorPalette/colorPalette.stories.tsx]: fixed

File: src/components/colorPalette/colorPalette.stories.tsx (hunk 3 above)
Comment (vitaliiRss) [reply src/components/colorPalette/colorPalette.stories.tsx]: fixed

File: src/components/colorPalette/colorPalette.module.scss (hunk 1 above)
Comment (vitaliiRss) [reply src/components/colorPalette/colorPalette.module.scss]: fixed

**Review by Abrosimov96 — APPROVED** [review]

File: src/components/colorPalette/colorPalette.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          <div className={s.title}>
+            {group.id}. {group.group}
+          </div>
+          <div className={s.colors}>
+            {group.groupColors.map(item => (
+              <div className={s.color} key={item.name}>
+                <div className={s.tile} style={{ backgroundColor: item.color }}></div>
+                <div className={s.info}>
+                  <div className={s.name}>{item.name}</div>
+                  <div className={s.colorHex}>{item.color}</div>
```
Comment (web-serge) [inline src/components/colorPalette/colorPalette.tsx:34]: There is an optional suggestion to add a color copy function
```
async function copyTextToClipboard (item.color) {
  try {
    await navigator.clipboard.writeText(text);
    console.log('Текст успешно скопирован в буфер обмена!');
  } catch (err) {
    console.error('Ошибка:', err);
  }
}
onClick={copyTextToClipboard }
```

**Review by web-serge — APPROVED** [review]

**Review by SergFediaev — APPROVED** [review]

File: src/components/colorPalette/colorPalette.tsx (hunk 4 above)
Comment (vitaliiRss) [reply src/components/colorPalette/colorPalette.tsx]: fixed

image

**Pull request merged**